library/L1_Peripheral/lpc40xx/watchdog.hpp

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <cstdint>
+#include <iterator>
+
+#include "L0_Platform/lpc40xx/LPC40xx.h"
+#include "L1_Peripheral/cortex/interrupt.hpp"
+#include "L1_Peripheral/lpc40xx/gpio.hpp"
+#include "L1_Peripheral/lpc40xx/watchdog_interface.hpp"
+#include "utility/time.hpp"
+#include "utility/bit.hpp"
+
+namespace sjsu
+{
+namespace lpc40xx
+{
+class Watchdog final : public sjsu::Watchdog
+{
+ public:
+  const uint8_t  kWatchdogResetAndEnable  = 0x3;
+  const uint32_t kTimerWarningMax = 0b11'1111'1111;
+  uint8_t timer_mode = 0b0000'0000;
+  inline static LPC_WDT_TypeDef * wdt_base = LPC_WDT;
+
+  static void WatchdogHandler()
+  {
+    // LED0 will flash upon entering the interrupt handler
+    Gpio led0(2, 3);
+    led0.SetDirection(sjsu::Gpio::Direction::kOutput);
+    led0.SetLow();
+  }
+
+  // This function converts seconds to units used by the watchdog
+  static uint32_t ConvertSeconds(uint32_t timer_variable)
+  {
+    // Every 125000 units written to the TC register is equal to 1 second

```

Comment:
In the datasheet it says that by it uses the watchdog clock source (Twdclk) with a frequency of 500 kHz. It says that the formula is: (Twdclk)*(Value written to the register)*(4). I was able to come up with 125000 units = 1 second by recording the amount of time it took for the watchdog interrupt to trigger. I tested it by passing in 10, 20, and 60 (seconds) into the function. I am still not sure what is the correct way to convert the value written to the register into seconds.

Reply:
Oh neat. So it does have its own clock source. It might be good to rename this from `ConvertSeconds()` (what does that even mean), to `ConvertDurationToWatchDogTicks()`. This gives you a better idea of what it is you are doing when you run this method.

You could probably do something like:
```suggestion
   static uint32_t ConvertDurationToWatchDogTicks(std::chrono::microseconds duration)
   { 
     // Every 125000 units written to the TC register is equal to 1 second
     return (duration.count() * 500_kHz) / 4;
   }
```

My suggestion above isn't the exact thing you should do. It won't compile. But you get the idea from looking at it. That's all you would have to do really.

Reply:
Ok, I renamed the method. I did something similar to that example, but now I am having trouble converting the std::chrono::microseconds object into a uint32_t type to write the TC register.

Reply:
Here is a sample:
```
std::chrono::microseconds duration = 10us;
duration.count();  // returns value 10.
```